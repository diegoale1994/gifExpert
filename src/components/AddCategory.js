import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState("");
    const handleInputText = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.length > 2) {
            setCategories(categories => [inputValue, ...categories]);
            setinputValue("");
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputText} />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


