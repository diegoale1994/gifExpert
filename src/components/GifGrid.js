import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__fadeIn"> {category} </h3>
            {loading && 'cargando'}
            <div className="card-grid">
                {images.map(img => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>
        </>
    )
}

GifGrid.prototypes = {
    category: PropTypes.string.isRequired
}