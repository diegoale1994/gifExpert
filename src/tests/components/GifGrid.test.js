import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('tests sobre gifgrid', () => {

    test('debe crearse el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category='slayer' />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se llama el hook', () => {
        const imgs = [{
            id: 123,
            title: 'aly',
            url: 'https://mock/jj.png'
        },
        {
            id: 125,
            title: 'alya',
            url: 'https://mock/jl.png'
        }]
        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category='slayer' />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    })

})
