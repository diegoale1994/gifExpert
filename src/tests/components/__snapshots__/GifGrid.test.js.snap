// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests sobre gifgrid debe crearse el componente 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
     
    slayer
     
  </h3>
  cargando
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`tests sobre gifgrid debe de mostrar items cuando se llama el hook 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
     
    slayer
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={123}
      key="123"
      title="aly"
      url="https://mock/jj.png"
    />
    <GifGridItem
      id={125}
      key="125"
      title="alya"
      url="https://mock/jl.png"
    />
  </div>
</Fragment>
`;
